### PR TITLE
feat: add cluster train script

### LINKED ISSUE #111: Set-up ability to run IPT using GPT-NeoX (already set-up in GPT-NeoX, configure to run from ChemNLP)

### LINKED ISSUE #120: Prepare chemrxiv dataset for initial GPT-NeoX finetuning tests

### PR DESCRIPTION
**Pull request by jackapbutler**
## Issues
* relates to #83 
* closes #111 
* closes #120 

## Details
This PR adds scripts for training chemnlp models on the Stability cluster + additional documentation. It also restructures the directory to have a root `experiments/` folder containing all scripts, configs and data related to cluster experiments.
* Update `experiments/EXPERIMENTS.md` with information about running slurm experiments.
* Adds `160M.yml`, `cluster_setup.yml` and `soft_prompt.yml` for model / training, cluster and peft configuration respectively.
* Adds a preparation script for the chemrxiv dataset which transforms it into a GPT-NeoX format.
* Updates some requirements in `env_creation.sh` which caused errors on the cluster.
* Adds `sbatch_run.sh` for running slurm experiments using chemnlp and GPT-NeoX

### CODE REVIEW

File: README.md
Comment (kjappelbaum) [inline README.md:1]: Do you use some Markdown formatter? If yes, we should perhaps all use the same one.

File: experiments/README.md
Comment (kjappelbaum) [inline experiments/README.md:1]: A very basic question we should perhaps also address there is how to get access to the clusters (I would simply say that contributors who think they need access should contact the core team)

File: experiments/configs/160M.yml
Comment (kjappelbaum) [inline experiments/configs/160M.yml:1]: i didn't look into the convention they use at Eleuther, but isn't the file extension misleading?

File: experiments/data/prepare_chemrxiv.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    # parse args
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "save_dir", help="Where you want to store the prepared dataset."
+    )
+    parser.add_argument(
+        "gptneox_dir", help="Where you can find the GPT-NeoX repository."
+    )
+    args = parser.parse_args()
+
```
Comment (kjappelbaum) [inline experiments/data/prepare_chemrxiv.py:27]: Minor point, should we agree on what framework we use for CLIs? (`click`, `fire`, simple `argparse`)

File: experiments/data/prepare_chemrxiv.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ]
+    save_path = f"{args.save_dir}/{DATASET}"
+    data_path = f"{save_path}/data.jsonl"
+    os.makedirs(save_path, exist_ok=True)
+    with jsonlines.open(data_path, "w") as writer:
+        writer.write_all(all_full_text_samples)
+
+    # execute gpt-neox processing
+    gpt_tool_path = f"{args.gptneox_dir}/tools/preprocess_data.py"
+    os.system(
```
Comment (kjappelbaum) [inline experiments/data/prepare_chemrxiv.py:41]: I guess the processing has no python API we could directly call?

File: experiments/scripts/stability-cluster/sbatch_run.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#! /bin/bash
+#SBATCH --job-name="chemtest"
+#SBATCH --nodes=1
+#SBATCH --ntasks-per-node=1
+#SBATCH --cpus-per-task=12
+# #SBATCH --gres=gpu:2
```
Comment (kjappelbaum) [inline experiments/scripts/stability-cluster/sbatch_run.sh:6]:
```suggestion
```

**Review by kjappelbaum — APPROVED** [review]
I must say I didn't run the code, but all the changes look reasonable and nice. Thanks!

File: experiments/scripts/stability-cluster/sbatch_run.sh (hunk 3 above)
Comment (jackapbutler) [reply experiments/scripts/stability-cluster/sbatch_run.sh:6]: what does this suggestion?

File: experiments/data/prepare_chemrxiv.py (hunk 2 above)
Comment (jackapbutler) [reply experiments/data/prepare_chemrxiv.py:41]: no, there doesn't seem to be, the main interface is just cli scripts

File: experiments/data/prepare_chemrxiv.py (hunk 1 above)
Comment (jackapbutler) [reply experiments/data/prepare_chemrxiv.py:27]: Ye I agree, I've used `click` before but don't really have any strong preference tbh

File: experiments/configs/160M.yml
Comment (jackapbutler) [reply experiments/configs/160M.yml:1]: Ye, they expect it to be in this format but I agree it's a bit odd

File: README.md
Comment (jackapbutler) [reply README.md:1]: I think I'm just using the default VSCode one but happy to use different extensions. Which one do you use?

**Pull request merged**